Sources/_TestingInternals/Discovery.cpp

```diff
@@ -366,12 +482,11 @@ static void enumerateTypeMetadataSections(const SectionEnumerator& body) {}
 #pragma mark -
 
 void swt_enumerateTypesWithNamesContaining(const char *nameSubstring, void *context, SWTTypeEnumerator body) {
-  enumerateTypeMetadataSections([=] (const void *section, size_t size) {
+  enumerateTypeMetadataSections([=] (const void *imageAddress, const void *section, size_t size, bool *stop) {
```

Comment:
Can we use `bool &` instead of `bool *` if this is meant to be non-null?

Reply:
This is just cribbed from what we do in Foundation for block-based enumeration.

Aside: it is not codified in any style guides anywhere, but I'm of the opinion that an `inout` or `out` argument should be a pointer so that at the call site you have to write `&` and within the callee you have to write `*` and it's clear to the caller that mutation may occur, whereas a mutable reference argument looks just like any other argument at the call site. We apply the same logic to `inout` in Swift, and we really wish this code were Swift. 🙃

Also we pass this parameter into Swift where there are no references, so it would be a bit confusing for it to be a reference here and a pointer there.